### PR TITLE
Remove Gson dependency from Leshan-core.

### PR DESCRIPTION
**Pull request by sbernard31**
Currently we use **gson** in `leshan-core` and for demos (`server-demo` and `bsserver-demo`)
And we use **minimal-json** for `leshan-cluster`.

We should limit us to only one Json library.

**minimal-json** force to write a little more code but it is faster, smaller, more flexible, and does not have dependency to java.sql.time (see #58)
So I replace **gson** by **minimal-json** in `leshan-core` which is the most important part.

Demos will still use **gson** :'(

### CODE REVIEW

File: leshan-core/src/main/java/org/eclipse/leshan/core/model/ObjectLoader.java [hunk 1]
```diff
@@ -92,11 +81,15 @@ public static ObjectModel loadDdfFile(InputStream input, String streamName) {
...
     public static List<ObjectModel> loadJsonStream(InputStream input) {
-        Reader reader = new InputStreamReader(input);
-        ObjectModel[] objectModels = GSON.fromJson(reader, ObjectModel[].class);
-        List<ObjectModel> models = new ArrayList<>();
-        Collections.addAll(models, objectModels);
-        return models;
+        try {
+            Reader reader = new InputStreamReader(input);
+            JsonValue json = Json.parse(reader);
+            List<ObjectModel> models = new ObjectModelSerDes().deserialize(json.asArray());
```
Comment (msangoi) [inline leshan-core/src/main/java/org/eclipse/leshan/core/model/ObjectLoader.java]: You can remove 'models' variable.

File: leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeEncoderTest.java [hunk 2]
```diff
@@ -189,7 +189,7 @@ public void json_encode_device_object_instance() {
         b.append("{\"n\":\"9\",\"v\":100},");
         b.append("{\"n\":\"10\",\"v\":15},");
         b.append("{\"n\":\"11\",\"v\":0},");
-        b.append("{\"n\":\"13\",\"v\":1367491215},");
+        b.append("{\"n\":\"13\",\"v\":1.367491215E9},");
```
Comment (msangoi) [inline leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeEncoderTest.java]: Don't we want to keep the value without exponent?

File: leshan-core/src/main/java/org/eclipse/leshan/core/model/ObjectLoader.java (hunk 1 above)
Comment (sbernard31) [reply leshan-core/src/main/java/org/eclipse/leshan/core/model/ObjectLoader.java]: I will do that.

File: leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeEncoderTest.java (hunk 2 above)
Comment (sbernard31) [reply leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeEncoderTest.java]: I agree this is strange to encode a date with this notation... But I think this respect the specification as in JSON, there is only one format for number called "Number" :p (see https://json.org/)

**Why this behave like this :**
When we encode we use several format :
LwM2mNode > LWM2M JSON  > JSON > String

The resource 13 is a date but we loose this information during conversion.
Date ==(leshan)==> Number ==(leshan)==> Double ==(minimal-json)==> String

Minimal-json use [Double.toString](https://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#toString(double))

We can imagine to use a custom minimal-json JsonWriter (it sounds not so good)
or trying to no convert all Number in Double. (we need to add some instanceof in JsonArrayEntrySerDes.jSerialize)

I don't know if it worth it?

Comment (msangoi) [reply leshan-core/src/test/java/org/eclipse/leshan/core/node/codec/LwM2mNodeEncoderTest.java]: Ok. indeed, the scientific notation is valid in Json. Let's keep it that way :)

**Pull request merged**